src/docs/stan-reference/examples.tex

```diff
@@ -7145,87 +7193,325 @@ \section{Fitting a Gaussian Process}\label{fit-gp.section}
+numerical issues that may arise in Stan when estimating a GP. 
+
+Perhaps most importantly, the parameters $\rho$ and $\alpha$ are weakly
+identified because of the equivalence of multivariate Gaussians with a
+covariance matrix parameterized by a Matern kernel. The ratio of the two
+parameters is well-identified, but in practice we put independent priors on the
+two hyperparameters because these two quantities are more interpretable than
+their ratio. 
+
+\subsubsection{Priors for length-scale}
+
+GPs are a flexible class of priors, and as such, can represent a wide spectrum
+of functions. Their flexibility can lead to numerical problems when estimating 
+a model in Stan using Euclidean Hamiltonian Monte Carlo. Length-scales that are
+smaller than the scale of the input data lead to a perfect fit with the
+observed data. This can require small step sizes when exploring this region
+of the parameter space. In order to limit the amount of posterior mass that
+resides in this region, the prior for length-scale should strongly regularize
+away from length-scales that are far smaller than the scale of the input.
+
+We may wish to put further soft constraints on the length-scale, but these are
+dependent on how the GP is used in our statistical model. 
+
+If our model consists of only the GP, i.e.:
+
+\begin{align}
+  f & \sim \distro{MultiNormal}\left(0, K_{\rho,\alpha}(x)\right) \\
+  y_i & \sim \distro{Normal}(f_i, \sigma) \, \forall i \in \{1, \dots, N\} \\
+  & x \in \reals^{N \times D}
+\end{align} 
+
+we likely don't need further constraints.  We'd like to allow the GP prior to
+represent both high-frequency and low-frequency functions, so our prior should
+put non-neglible mass on both sets of functions. In this case, an inverse gamma
+will work well provided it has a left tail that puts neglible mass on
+length-scales that are small with respect to the input data.  
+
+If we're using the GP as a component in a larger model that includes an overall
+mean and fixed effects for the same variables we're using as the domain for the
+GP, i.e.:
+
+\begin{align}
+  f & \sim \distro{MultiNormal}\left(0, K_{\rho,\alpha}(x)\right) \\
+  y_i & \sim \distro{Normal}(\beta_0 + x_i \beta_{[1:D]} + f_i, \sigma) \, \forall i \in \{1, \dots, N\} \\
+  & x_i^T, \beta_{[1:D]} \in \reals{R}^D,\, x \in \reals^{N \times D}, f \in \
+\end{align} 
+
+we'll likely want to constrain large length-scales as well.  A length scale
+that is larger than the scale of the data yields a GP posterior that is
+practically linear (with respect to the particular covariate) and increasing
+the length scale has little impact on the likelihood. This will introduce
+nonidentiability in our model, as both the fixed effects and the GP will
+explain similar variation. In order to limit the amount of overlap between the
+GP and the linear regression, we should use a prior with a sharper right tail
+to limit the GP to higher-frequency functions. We can use an inverse Gaussian,
```

Comment:
Is the inverse Gaussian (I'm assuming it's this: https://en.wikipedia.org/wiki/Inverse_Gaussian_distribution) available in Stan? (I wanted to use it but didn't find it)